### PR TITLE
[GEOS-10632] Make sure GetLegendGraphics honors the WMS memory service limits

### PR DESCRIPTION
**Pull request by aaime**
[GEOS-10632](https://osgeo-org.atlassian.net/browse/GEOS-10632)

Supercedes https://github.com/geoserver/geoserver/pull/6130

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetLegendGraphicTest.java [hunk 1]
```diff
@@ -78,6 +86,18 @@ protected void onSetUp(SystemTestData testData) throws Exception {
                 defaultWorkspace, "wsCustom", "point_test.sld", getClass(), catalog, legend);
     }
 
+    @After
+    public void cleaMemoryLimit() throws Exception {
```
Comment (dromagnoli) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetLegendGraphicTest.java:90]:
```suggestion
    public void cleanMemoryLimit() throws Exception {
```

File: src/wms/src/test/java/org/geoserver/wms/legendgraphic/ImageQueueTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        // When
+        imageQueue.add(renderedImage);
+
+        // Then
+        assertEquals(1, imageQueue.get().size());
+    }
+
+    @Test(expected = ServiceException.class)
+    public void testExceedingMaxMemoryWhenAddingToTallieQueue() {
```
Comment (dromagnoli) [inline src/wms/src/test/java/org/geoserver/wms/legendgraphic/ImageQueueTest.java:39]:
```suggestion
    public void testExceedingMaxMemoryWhenAddingToTalliedQueue() {
```

Comment (dromagnoli) [reply src/wms/src/test/java/org/geoserver/wms/legendgraphic/ImageQueueTest.java:39]: Not sure if it's better Tallied or Tally

**Review by dromagnoli — APPROVED** [review]
Looks good to me

File: src/wms/src/test/java/org/geoserver/wms/legendgraphic/ImageQueueTest.java (hunk 2 above)
Comment (aaime) [reply src/wms/src/test/java/org/geoserver/wms/legendgraphic/ImageQueueTest.java:39]: Oh right, I have to rename the test (the ImageQueue used to be called TalliedQueue)

**Pull request merged**